### ISSUE TITLE
Repo is reset after every push when using ssh type git url

### ISSUE THREAD

**Issue by jxltom** [dokku/dokku#2994]
Since the SCP like git url does not support custom port, so I have to use ```ssh://dokku@domain:port/appname``` format git url. But when using this kind url, the remote repo in ```/home/dokku/myapp/``` will reset for every new push. This means all content within this directory is cleaned up and only ```cache``` folder is left. So I can not set environment variable for the application since ```ENV``` file will also be removed every time.

I found out I have to use ```ssh://dokku@domain:port/appname.git``` format git url which must ends with ```.git```. But in this way, the domain of the application is totally broken.

Output of the following commands

```shell
-----> uname: Linux  3.13.0-32-generic #57-Ubuntu SMP Tue Jul 15 03:51:08 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
-----> memory: 
             total       used       free     shared    buffers     cached
Mem:          3953       3529        423         18        256       2235
-/+ buffers/cache:       1037       2916
Swap:            0          0          0
-----> docker version: 
Client:
 Version:      17.11.0-ce
 API version:  1.34
 Go version:   go1.8.3
 Git commit:   1caf76c
 Built:        Mon Nov 20 18:36:37 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.11.0-ce
 API version:  1.34 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   1caf76c
 Built:        Mon Nov 20 18:35:10 2017
 OS/Arch:      linux/amd64
 Experimental: false
-----> docker daemon info: 
Containers: 8
 Running: 2
 Paused: 0
 Stopped: 6
Images: 12
Server Version: 17.11.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 32
 Dirperm1 Supported: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 992280e8e265f491f7a624ab82f3e238be086e49
runc version: 0351df1c5a66838d0c392b4ac4cf9450de844e2d
init version: 949e6fa
Security Options:
 apparmor
Kernel Version: 3.13.0-32-generic
Operating System: Ubuntu 14.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.861GiB
Name: iZ621ofqrmvZ
ID: Q3LP:5ZLX:C5TC:O3UT:X7RI:H6TM:IDQS:L6JR:KOSA:KDKM:ILVY:G5ZZ
Docker Root Dir: /var/lib/docker
Debug Mode (client): true
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
-----> sigil version: 0.4.0
-----> herokuish version: 
herokuish: 0.3.32
buildpacks:
  heroku-buildpack-multi     v1.0.0
  heroku-buildpack-ruby      v168
  heroku-buildpack-nodejs    v99
  heroku-buildpack-clojure   v77
  heroku-buildpack-python    v99
  heroku-buildpack-java      v53
  heroku-buildpack-gradle    v23
  heroku-buildpack-scala     v78
  heroku-buildpack-play      v26
  heroku-buildpack-php       v122
  heroku-buildpack-go        v72
  heroku-buildpack-erlang    fa17af9
  buildpack-nginx            v8
-----> dokku version: 0.10.5
-----> dokku plugins: 
plugn: 0.3.0
  00_dokku-standard    0.10.5 enabled    dokku core standard plugin
  20_events            0.10.5 enabled    dokku core events logging plugin
  apps                 0.10.5 enabled    dokku core apps plugin
  build-env            0.10.5 enabled    dokku core build-env plugin
  certs                0.10.5 enabled    dokku core certificate management plugin
  checks               0.10.5 enabled    dokku core checks plugin
  common               0.10.5 enabled    dokku core common plugin
  config               0.10.5 enabled    dokku core config plugin
  docker-options       0.10.5 enabled    dokku core docker-options plugin
  domains              0.10.5 enabled    dokku core domains plugin
  enter                0.10.5 enabled    dokku core enter plugin
  git                  0.10.5 enabled    dokku core git plugin
  logs                 0.10.5 enabled    dokku core logs plugin
  named-containers     0.10.5 enabled    dokku core named containers plugin
  nginx-vhosts         0.10.5 enabled    dokku core nginx-vhosts plugin
  plugin               0.10.5 enabled    dokku core plugin plugin
  proxy                0.10.5 enabled    dokku core proxy plugin
  ps                   0.10.5 enabled    dokku core ps plugin
  repo                 0.10.5 enabled    dokku core repo plugin
  shell                0.10.5 enabled    dokku core shell plugin
  ssh-keys             0.10.5 enabled    dokku core ssh-keys plugin
  storage              0.10.5 enabled    dokku core storage plugin
  tags                 0.10.5 enabled    dokku core tags plugin
  tar                  0.10.5 enabled    dokku core tar plugin
```

**Comment by jxltom** [comment]
Actually I found out a temporary solution by add ssh alias in ```~/.ssh/config``` and add ssh remote with custom port. In this way, I can push by ```git push ssh_alias:myapp``` and everything works fine.

**Comment by josegonzalez** [comment]
Fixed by #2926 and released in v0.11.0.
